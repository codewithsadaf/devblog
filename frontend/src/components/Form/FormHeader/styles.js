//imports
import styled, { css } from "styled-components";
//components styles
import { TextComponent } from "../../Layout/TextComponent";

export const Container = styled.div`
    ${({ theme }) => css`
        margin-bottom: 1em;
    `}
`;

export const Text = styled(TextComponent)`
    margin-top: 0.3em;
`;
