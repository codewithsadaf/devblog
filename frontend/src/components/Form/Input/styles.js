//imports
import styled, { css } from "styled-components";

export const FormControl = styled.div`
    display: flex;
    flex-direction: column;
    margin-bottom: 1em;
`;

export const LabelComponent = styled.label`
    display: block;
    margin-bottom: 0.3em;
    font-weight: 700;
`;

export const InputComponent = styled.input`
    ${({ theme }) => css`
        padding: 7px 0;
        outline: none;
        border: none;
        border-bottom: 1px solid ${theme.colors.gray};
        background-color: transparent;
        font-size: ${theme.font.sizes.default};
        transition: ${theme.transition.tran_05};

        &:focus {
            border-bottom: 1px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `}
`;
