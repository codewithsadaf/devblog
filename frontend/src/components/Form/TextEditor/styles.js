//imports
import styled from "styled-components";

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    margin: 1em 0;
`;

export const Label = styled.span`
    display: block;
    margin-bottom: 0.3em;
    font-weight: 700;
`;
