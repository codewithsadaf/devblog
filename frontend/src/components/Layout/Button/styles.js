//imports
import styled, { css } from "styled-components";

const handleButtonVariant = {
    default: (theme) => css`
        background: ${theme.colors.primaryColor};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: transparent;
            color: ${theme.colors.primaryColor};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,
    outline: (theme) => css`
        background: transparent;
        color: ${theme.colors.primaryColor};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: ${theme.colors.primaryColor};
            color: ${theme.colors.white};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,

    success: (theme) => css`
        background: ${theme.colors.success};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.success};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,

    danger: (theme) => css`
        background: ${theme.colors.danger};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.danger};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,
};

const handleUppercase = () => css`
    text-transform: uppercase;
`;

const handleMaxWidth = () => css`
    width: 100%;
    padding: 1em 0;
`;

const handleSmallStyles = (theme) => css`
    padding: 7px 30px;
    border-radius: 4px;
    font-size: ${theme.font.sizes.small};
    border: 1px solid;

    &:hover {
        border: 1px solid;
    }
`;

export const ButtonContainer = styled.button`
    ${({ theme, buttonVariant, uppercase, maxWidth, isSmall }) => css`
        padding: 0.5em 1.8em;
        font-size: 1em;
        border: none;
        font-weight: bold;
        cursor: pointer;
        ${handleButtonVariant[buttonVariant](theme)};
        ${uppercase && handleUppercase()};
        ${maxWidth && handleMaxWidth()};
        ${isSmall && handleSmallStyles(theme)};
    `}
`;
