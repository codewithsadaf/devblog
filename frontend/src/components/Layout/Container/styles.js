//imports
import styled from "styled-components";

export const Container = styled.div`
    min-height: 60vh;
    margin-bottom: 5em;
    padding: 0 2em;
`;
