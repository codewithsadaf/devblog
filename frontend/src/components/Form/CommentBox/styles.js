//imports
import styled, { css } from "styled-components";
//components styles
import { InputComponent } from "../Input/styles";
import { ButtonContainer as Button } from "../../Layout/Button/styles";

export const Container = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;

    & ${Button} {
        padding: 12px 16px;
    }
`;

export const CommentInput = styled(InputComponent)`
    ${({ theme }) => css`
        border: 1px solid ${theme.colors.gray};
        transition: ${theme.transition.tran_03};
        padding: 12px 7px;
        width: 90%;
        border-radius: 5px;
        font-size: 1em;
        margin-right: 7px;

        &:focus {
            border: 1px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `}
`;
