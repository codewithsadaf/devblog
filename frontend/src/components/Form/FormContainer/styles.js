//imports
import styled, { css } from "styled-components";
//components styles
import { ButtonContainer as Button } from "../../../components/Layout/Button/styles";

export const Container = styled.div`
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const Form = styled.form`
    ${({ theme }) => css`
        width: 450px;
        margin-top: 3em;

        & > ${Button} {
            margin-top: 1em;
        }

        @media ${theme.media.lteMedium} {
            width: 100%;
            min-width: 0;
            max-width: 100%;
        }
    `}
`;
