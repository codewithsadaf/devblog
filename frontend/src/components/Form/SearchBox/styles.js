//imports
import styled, { css } from "styled-components";
//components styles
import { ButtonContainer as Button } from "../../Layout/Button/styles";
import { Container as Heading } from "../../Layout/Heading/styles";
import { InputComponent } from "../Input/styles";

export const Container = styled.div`
    width: 100%;
    margin: 0 auto;
    margin-bottom: 2em;

    & ${Heading} {
        margin-bottom: 0.5em;
    }
`;

export const SearchWrapper = styled.div`
    ${({ theme }) => css`
        display: flex;
        justify-content: space-between;
        width: 90%;
        margin: 0 auto;

        & ${Button} {
            width: 20%;
            padding: 7px;
            border-radius: 2px;

            @media ${theme.media.lteMedium} {
                width: 35%;
            }
        }

        @media ${theme.media.lteMedium} {
            width: 100%;
        }
    `}
`;

export const SearchInput = styled(InputComponent)`
    ${({ theme }) => css`
        width: 78%;

        @media ${theme.media.lteMedium} {
            width: 60%;
        }
    `}
`;
