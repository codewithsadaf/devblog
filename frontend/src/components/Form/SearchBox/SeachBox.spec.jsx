//imports
import { fireEvent, screen } from "@testing-library/react";
//styles
import { renderTheme } from "../../../styles/render-theme";
import { theme } from "../../../styles/theme";
//components
import { SearchBox } from "./";

describe("<SearchBox />", () => {
    const mockFnChange = jest.fn();
    const mockFnClick = jest.fn();

    beforeEach(() => {
        renderTheme(
            <SearchBox OnChangeFn={mockFnChange} OnClickFn={mockFnClick} />
        );
    });

    describe("Rendering", () => {
        it("should render SearchBox", () => {
            const heading = screen.getByRole("heading", {
                name: "Veja os nossos posts mais recentes",
            });

            expect(heading).toBeInTheDocument();
        });

        it("should render SearchWrapper component with 90% of width", () => {
            const buttonElem = screen.getByRole("button", {
                name: "Pesquisar",
            });

            expect(buttonElem.parentElement).toHaveStyleRule("width", "90%");
        });

        it("should render SearchWrapper component with 100% of width on mobile ", () => {
            const buttonElem = screen.getByRole("button", {
                name: "Pesquisar",
            });

            expect(buttonElem.parentElement).toHaveStyleRule("width", "100%", {
                media: theme.media.lteMedium,
            });
        });

        it("should render button component with 20% of width ", () => {
            const buttonElem = screen.getByRole("button", {
                name: "Pesquisar",
            });

            expect(buttonElem).toHaveStyle("width: 20%;");
        });

        it("should render input component with 78% of width", () => {
            const inputElement = screen.getByPlaceholderText(
                "Pesquisar por tags..."
            );

            expect(inputElement).toHaveStyleRule("width", "78%");
        });

        it("should render input component with 60% of width on mobile", () => {
            const inputElement = screen.getByPlaceholderText(
                "Pesquisar por tags..."
            );

            expect(inputElement).toHaveStyleRule("width", "60%", {
                media: theme.media.lteMedium,
            });
        });

        it("should match snapshot", () => {
            const { container } = renderTheme(
                <SearchBox OnChangeFn={mockFnChange} OnClickFn={mockFnClick} />
            );

            expect(container.firstChild).toMatchSnapshot();
        });
    });

    describe("Functionality", () => {
        it("should call on change function when input is typed", () => {
            const inputElement = screen.getByPlaceholderText(
                "Pesquisar por tags..."
            );

            fireEvent.change(inputElement, { target: { value: "test" } });

            expect(mockFnChange).toHaveBeenCalledTimes(1);
        });
    });

    describe("Functionality", () => {
        it("should call on click function when button is clicked", () => {
            const buttonElem = screen.getByRole("button", {
                name: "Pesquisar",
            });

            fireEvent.click(buttonElem);

            expect(mockFnClick).toHaveBeenCalledTimes(1);
        });
    });
});
