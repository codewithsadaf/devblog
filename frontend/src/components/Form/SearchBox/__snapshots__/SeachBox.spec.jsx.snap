// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> Rendering should match snapshot 1`] = `
.c7 {
  padding: 0.5em 1.8em;
  font-size: 1em;
  border: none;
  font-weight: bold;
  cursor: pointer;
  background: #000;
  color: #fff;
  border: 2px solid #000;
  border-radius: 4px;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c7:hover {
  background: transparent;
  color: #000;
  border: 2px solid #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 {
  text-transform: none;
  font-weight: bold;
  font-size: 2em;
  line-height: 110%;
}

.c4 {
  padding: 7px 0;
  outline: none;
  border: none;
  border-bottom: 1px solid #ccc;
  background-color: transparent;
  font-size: 1em;
  -webkit-transition: all 500ms ease-in-out;
  transition: all 500ms ease-in-out;
}

.c4:focus {
  border-bottom: 1px solid #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 {
  width: 100%;
  margin: 0 auto;
  margin-bottom: 2em;
}

.c0 .c1 {
  margin-bottom: 0.5em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 90%;
  margin: 0 auto;
}

.c3 .c6 {
  width: 20%;
  padding: 7px;
  border-radius: 2px;
}

.c5 {
  width: 78%;
}

@media (max-width:768px) {
  .c3 .c6 {
    width: 35%;
  }
}

@media (max-width:768px) {
  .c3 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c5 {
    width: 60%;
  }
}

<div
  class="c0"
>
  <h3
    class="c1 c2"
  >
    Veja os nossos posts mais recentes
  </h3>
  <div
    class="c3"
  >
    <input
      class="c4 c5"
      placeholder="Pesquisar por tags..."
      type="text"
    />
    <button
      class="c6 c7"
      type="button"
    >
      Pesquisar
    </button>
  </div>
</div>
`;
